### PR TITLE
Remove 405 from render method 

### LINKED ISSUE #8045: [v9 regression] Can't render a page as a response to POST anymore
# Bug report

## Describe the bug

Hi! 
Since Next.js v9, we are no longer able to render a page as a response to POST request anymore. Basically, everything seems to be going fine on the server, but when it comes to rendering the page, Next throws a 405 error. We believe it's due to this line of code: https://github.com/zeit/next.js/blob/fce7e4eaee45e89557ceaa81fbd42057f8596e22/packages/next-server/server/next-server.ts#L509

## To Reproduce

Here's an example of a flow that leads to the error. We are using custom Express server, and serving static nojs pages which can only use forms to submit data, and we don't keep any server-side state.

1. `GET` to `/nojs/settings` -> handled by `server.get("/nojs/settings"...)` -> renders `/nojs/settings` page.
2. To change settings, user `POST`s a form to `/nojs/settings` -> handled by `server.post("/nojs/settings"...)`
3. Server needs to respond with the page that has some additional data (error messages etc.) passed as `queryParams`. So after whatever needs to be done to get that data, `server.post("/nojs/settings"...)` is doing `app.render(req, res, "/nojs/settings", { data })`
4. User is going to see the `_error.js` page with `405` error code.

## Expected behavior

Before v9, `app.render()` on `POST` request used to render a page without any problems. 
Support for this behaviour is needed to serve static nojs pages, that only use forms to submit data, without introducing server-side state.
This is a major regression for this rather standard web practice use case and because of that we're stuck back on v8.

### PR DESCRIPTION
**Pull request by huv1k**
This PR reverts changes of `405` on different `http` methods then `GET` and `HEAD`

Fixes #8045

### CODE REVIEW

File: test/integration/api-support/test/index.test.js [hunk 1]
```diff
@@ -197,14 +197,6 @@ function runTests (serverless = false) {
     expect(data).toEqual({ nextjs: 'cool' })
   })
 
-  it('should return 405 on POST on pages', async () => {
```
Comment (timneutkens) [inline test/integration/api-support/test/index.test.js]: Let's update these tests to check if it returns 200

**Comment by ijjk** [comment]
## Stats from current PR

<details>
<summary>Click to expand stats</summary>

| | zeit/next.js canary | huv1k/next.js post-fix | Change |
| - | - | - | - |
| Build Duration | 23.6s | 23.5s | -109ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | -1.03 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.39 kB | 2.39 kB | ✓ |
| Client `_app` gzip Size | 1.08 kB | 1.08 kB | ✓ |
| Client `_app` Modern Size | 1.83 kB | 1.83 kB | ✓ |
| Client `_app` gzip Modern Size | 890 B | 890 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | ✓ |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 301 B | ✓ |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Base Rendered Size | 2.76 kB | 2.76 kB | ✓ |
| Build Dir Size | 1.44 MB | 1.44 MB | ✓ |

</details>


<details>
<summary>Click to expand serverless stats</summary>

| | zeit/next.js canary | huv1k/next.js post-fix | Change |
| - | - | - | - |
| Build Duration | 25.3s | 24.8s | -567ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | -1.03 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | -1 B |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.39 kB | 2.39 kB | ✓ |
| Client `_app` gzip Size | 1.08 kB | 1.08 kB | -1 B |
| Client `_app` Modern Size | 1.83 kB | 1.83 kB | ✓ |
| Client `_app` gzip Modern Size | 890 B | 890 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | -1 B |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 300 B | -1 B |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Serverless `pages/link` Size | 256 kB | 256 kB | ✓ |
| Serverless `pages/link` gzip Size | 68.8 kB | 68.8 kB | ⚠️  +1 B |
| Serverless `pages/index` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/index` gzip Size | 66.6 kB | 66.6 kB | ⚠️  +1 B |
| Serverless `pages/_error` Size | 248 kB | 248 kB | ✓ |
| Serverless `pages/_error` gzip Size | 66.3 kB | 66.3 kB | ⚠️  +1 B |
| Serverless `pages/routerDirect` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/routerDirect` gzip Size | 66.5 kB | 66.5 kB | ⚠️  +1 B |
| Serverless `pages/withRouter` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/withRouter` gzip Size | 66.6 kB | 66.6 kB | ⚠️  +1 B |
| Build Dir Size | 2.65 MB | 2.65 MB | ✓ |

</details>

**Comment by ijjk** [comment]
## Stats from current PR

<details>
<summary>Click to expand stats</summary>

| | zeit/next.js canary | huv1k/next.js post-fix | Change |
| - | - | - | - |
| Build Duration | 24s | 23.8s | -195ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | -1.03 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.39 kB | 2.39 kB | ✓ |
| Client `_app` gzip Size | 1.08 kB | 1.08 kB | ✓ |
| Client `_app` Modern Size | 1.83 kB | 1.83 kB | ✓ |
| Client `_app` gzip Modern Size | 890 B | 890 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | ✓ |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 301 B | ✓ |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Base Rendered Size | 2.76 kB | 2.76 kB | ✓ |
| Build Dir Size | 1.44 MB | 1.44 MB | ✓ |

</details>


<details>
<summary>Click to expand serverless stats</summary>

| | zeit/next.js canary | huv1k/next.js post-fix | Change |
| - | - | - | - |
| Build Duration | 26.3s | 25.9s | -444ms |
| `node_modules` Size | 41.6 MB | 41.6 MB | -1.03 kB |
| Total Bundle (main, webpack, commons) Size | 214 kB | 214 kB | ✓ |
| Total Bundle (main, webpack, commons) gzip Size | 70.6 kB | 70.6 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern Size | 189 kB | 189 kB | ✓ |
| Total Bundle (main, webpack, commons) Modern gzip Size | 62.4 kB | 62.4 kB | ✓ |
| Client `_app` Size | 2.39 kB | 2.39 kB | ✓ |
| Client `_app` gzip Size | 1.08 kB | 1.08 kB | ✓ |
| Client `_app` Modern Size | 1.83 kB | 1.83 kB | ✓ |
| Client `_app` gzip Modern Size | 890 B | 890 B | ✓ |
| Client `_error` Size | 8.22 kB | 8.22 kB | ✓ |
| Client `_error` gzip Size | 3.16 kB | 3.16 kB | ✓ |
| Client `_error` Modern Size | 5.85 kB | 5.85 kB | ✓ |
| Client `_error` gzip Modern Size | 2.33 kB | 2.33 kB | ✓ |
| Client `pages/index` Size | 343 B | 343 B | ✓ |
| Client `pages/index` gzip Size | 246 B | 246 B | ✓ |
| Client `pages/index` Modern Size | 319 B | 319 B | ✓ |
| Client `pages/index` gzip Modern Size | 254 B | 254 B | ✓ |
| Client `pages/link` Size | 4.18 kB | 4.18 kB | ✓ |
| Client `pages/link` gzip Size | 1.82 kB | 1.82 kB | ✓ |
| Client `pages/link` Modern Size | 3.81 kB | 3.81 kB | ✓ |
| Client `pages/link` gzip Modern Size | 1.72 kB | 1.72 kB | ✓ |
| Client `pages/routerDirect` Size | 423 B | 423 B | ✓ |
| Client `pages/routerDirect` gzip Size | 306 B | 306 B | ✓ |
| Client `pages/routerDirect` Modern Size | 411 B | 411 B | ✓ |
| Client `pages/routerDirect` gzip Modern Size | 314 B | 314 B | ✓ |
| Client `pages/withRouter` Size | 435 B | 435 B | ✓ |
| Client `pages/withRouter` gzip Size | 301 B | 301 B | ✓ |
| Client `pages/withRouter` Modern Size | 423 B | 423 B | ✓ |
| Client `pages/withRouter` gzip Modern Size | 309 B | 309 B | ✓ |
| Client `main` Size | 16.3 kB | 16.3 kB | ✓ |
| Client `main` gzip Size | 5.73 kB | 5.73 kB | ✓ |
| Client `main` Modern Size | 13.1 kB | 13.1 kB | ✓ |
| Client `main` Modern gzip Size | 4.99 kB | 4.99 kB | ✓ |
| Client `commons` Size | 195 kB | 195 kB | ✓ |
| Client `commons` gzip Size | 63.6 kB | 63.6 kB | ✓ |
| Client `commons` Modern Size | 176 kB | 176 kB | ✓ |
| Client `commons` Modern gzip Size | 57.4 kB | 57.4 kB | ✓ |
| Client `webpack` Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` gzip Size | 778 B | 778 B | ✓ |
| Client `webpack` Modern Size | 1.53 kB | 1.53 kB | ✓ |
| Client `webpack` Modern gzip Size | 785 B | 785 B | ✓ |
| Serverless `pages/link` Size | 256 kB | 256 kB | ✓ |
| Serverless `pages/link` gzip Size | 68.8 kB | 68.8 kB | -1 B |
| Serverless `pages/index` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/index` gzip Size | 66.6 kB | 66.6 kB | -1 B |
| Serverless `pages/_error` Size | 248 kB | 248 kB | ✓ |
| Serverless `pages/_error` gzip Size | 66.3 kB | 66.3 kB | -2 B |
| Serverless `pages/routerDirect` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/routerDirect` gzip Size | 66.5 kB | 66.5 kB | ✓ |
| Serverless `pages/withRouter` Size | 249 kB | 249 kB | ✓ |
| Serverless `pages/withRouter` gzip Size | 66.6 kB | 66.6 kB | -2 B |
| Build Dir Size | 2.65 MB | 2.65 MB | ✓ |

</details>

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thanks @huv1k 🙏

**Pull request merged**